File: backend/dalathasfarm/src/main/java/com/example/dalathasfarm/responses/orderdetail/OrderDetailResponse.java
package com.example.dalathasfarm.responses.orderdetail;

import com.example.dalathasfarm.models.OrderDetail;
import com.example.dalathasfarm.responses.product.ProductResponse;
import lombok.*;

import java.math.BigDecimal;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class OrderDetailResponse {
    private Integer orderId;
    private ProductResponse productResponse;
    private Integer quantity;
    private BigDecimal price;
    private BigDecimal totalMoney;

    public static OrderDetailResponse fromOrderDetail(OrderDetail orderDetail)
    {
        return OrderDetailResponse.builder()
                .orderId(orderDetail.getOrder().getId())
                .productResponse(ProductResponse.fromProduct(orderDetail.getProduct()))
                .quantity(orderDetail.getQuantity())
                .price(orderDetail.getPrice())
                .totalMoney(orderDetail.getTotalMoney())
                .build();
    }
}

File: android/lib/responses/order_detail_response.dart
import 'package:android/models/product.dart';
import 'package:decimal/decimal.dart';

class OrderDetailResponse{

  final int orderId;
  final Product productResponse;
  final int quantity;
  final double price;
  final double totalMoney;

  OrderDetailResponse({
    required this.orderId,
    required this.productResponse,
    required this.quantity,
    required this.price,
    required this.totalMoney,
  });

  factory OrderDetailResponse.fromJson(Map<String, dynamic> json) {
    return OrderDetailResponse(
      orderId: json['orderId'],
      productResponse: Product.fromJson(json['productResponse']),
      quantity: json['quantity'],
      price: (json['price'] as num).toDouble(),
      totalMoney: (json['totalMoney'] as num).toDouble(),
    );
  }
}